### DISCUSSION TITLE
Next.Js | MultiWindow | Open Links Support

### DISCUSSION THREAD

**Discussion by dfirence** [tauri-apps/tauri#1163] [Q&A] [answered]
Hi Awesome Tauri Team

quick questions please

What is the possible ETA of MultiWindow Support in Tauri? My use case is for multi-window support and I would want to avoid using Electron.

Does Tauri currently support NextJS or does it work with NextJS?

And lastly, do you have a code sample that shows how have a Tauri App coordinate a link being opened in the default browser of the machine?

**Comment by nothingismagick** [comment]
Mulltiwindow depends a bit on some internal revisions and upstream issues. No delivery date possible, but hopefully Soon :tm:.

NextJS requires a backend, and our backend is rust. There is however, the possibility to use a sidecar, [please see this page](https://tauri.studio/en/docs/usage/recipes/bundler/sidecar).

I don't understand your final question. Do you mean a link in a tauri webview is clicked and you want that to open a specific page in the browser? If so, you can just use the `target="_blank"` notation in an `<a>` element in your html.

**Comment by nklayman** [comment, marked as answer]
We have multi window support working in the dev branch, and it will be available in the next release. Tauri can't really support Next.js as it requires a server, but if you can use [Next export](https://nextjs.org/docs/advanced-features/static-html-export), then it will work fine. You could also load the content from a remote server instead of shipping it with your tauri app.

**Reply by philipaarseth** [reply]
So Nextjs with SSG is fine. IS adding a SSR compatibility layer/ support something Tauri would consider down the line?

**Reply by JonasKruckenberg** [reply]
The [`mocks`](https://tauri.studio/docs/api/js/modules/mocks) module exported by `@tauri-apps/api` can be used to fake a Tauri environment. This includes SSG, so you can collect IPC requests during SSG using the `mockIPC` function.

This is pretty manual work right now, but it's the only way to support so many different environments.

SSR support (as-in "Run the full NextJS server in the rust core") is not planned and is something we all highly advise against. I would also say NextJS is a poor fit for a desktop app, to begin with since the route-based layout doesn't map that well to highly dynamic desktop apps, but that is your choice in the end.

**Comment by nothingismagick** [comment]
But why ship a server if tauri can support not having one? I just don't see the utility.

**Comment by 0xle0ne** [comment]
Hey there just in case people wonder there is a showcase example on how to use nextjs with tauri [there](https://github.com/leon3s/nextauri)

**Comment by rakeshbhai0091** [comment]
how to make make multi window app

i using tauri with nextjs 

how to make on button click to open google in tauri app with custom headers or etc
menage urls also from nextjs
